SimCalorimetry/HGCalSimProducers/src/HGCDigitizer.cc

```diff
@@ -218,38 +228,72 @@ HGCDigitizer::HGCDigitizer(const edm::ParameterSet& ps,
   }
 
   if(hitCollection_.find("HitsEE")!=std::string::npos) {
-    mySubDet_=ForwardSubdetector::HGCEE;
+    if (geometryType_ == 0) {
+      mySubDet_=ForwardSubdetector::HGCEE;
+    } else {
+      myDet_   =DetId::HGCalEE;
+    }
     theHGCEEDigitizer_=std::unique_ptr<HGCEEDigitizer>(new HGCEEDigitizer(ps) );
   }
-  if(hitCollection_.find("HitsHEfront")!=std::string::npos)
-    {
+  if(hitCollection_.find("HitsHEfront")!=std::string::npos) {
+    if (geometryType_ == 0) {
       mySubDet_=ForwardSubdetector::HGCHEF;
-      theHGCHEfrontDigitizer_=std::unique_ptr<HGCHEfrontDigitizer>(new HGCHEfrontDigitizer(ps) );
+    } else {
+      myDet_   =DetId::HGCalHSi;
     }
-  if(hitCollection_.find("HcalHits")!=std::string::npos)
-    {
+    theHGCHEfrontDigitizer_=std::unique_ptr<HGCHEfrontDigitizer>(new HGCHEfrontDigitizer(ps) );
+  }
+  if(hitCollection_.find("HcalHits")!=std::string::npos) {
+    if (geometryType_ == 0) {
       mySubDet_=ForwardSubdetector::HGCHEB;
-      theHGCHEbackDigitizer_=std::unique_ptr<HGCHEbackDigitizer>(new HGCHEbackDigitizer(ps) );
+    } else {
+      myDet_   =DetId::HGCalHSc;
     }
+    theHGCHEbackDigitizer_=std::unique_ptr<HGCHEbackDigitizer>(new HGCHEbackDigitizer(ps) );
+  }
+  if(hitCollection_.find("HitsHEback")!=std::string::npos) {
+    if (geometryType_ == 0) {
+      mySubDet_=ForwardSubdetector::HGCHEB;
+    } else {
+      myDet_   =DetId::HGCalHSc;
+    }
+    theHGCHEbackDigitizer_=std::unique_ptr<HGCHEbackDigitizer>(new HGCHEbackDigitizer(ps) );
+  }
 }
 
 //
 void HGCDigitizer::initializeEvent(edm::Event const& e, edm::EventSetup const& es)
 {
   // reserve memory for a full detector
   unsigned idx = std::numeric_limits<unsigned>::max();
-  switch(mySubDet_) {
-  case ForwardSubdetector::HGCEE:
-    idx = 0;
-    break;
-  case ForwardSubdetector::HGCHEF:
-    idx = 1;
-    break;
-  case ForwardSubdetector::HGCHEB:
-    idx = 2;
-    break;
-  default:
-    break;
+  if (geometryType_ == 0) {
```

Comment:
this pattern is repeated in `finalizeEvent` - it should be a separate function to avoid duplication